pkg/firewall/firewall_test.go

```diff
@@ -1,162 +1,383 @@
 package firewall
 
 import (
+	"fmt"
 	"github.com/keep-network/keep-core/pkg/chain/local_v1"
 	"github.com/keep-network/keep-core/pkg/operator"
-	"math/big"
+	"reflect"
 	"testing"
 	"time"
 
 	"github.com/keep-network/keep-common/pkg/cache"
 )
 
-var minimumStake = big.NewInt(1000)
-var cachingPeriod = time.Second
+const cachingPeriod = time.Second
 
-func TestHasMinimumStake(t *testing.T) {
-	stakeMonitor := local_v1.NewStakeMonitor(minimumStake)
-	policy := &minimumStakePolicy{
-		stakeMonitor:        stakeMonitor,
+func TestValidate_OperatorNotRecognized_NoApplications(t *testing.T) {
+	policy := &anyApplicationPolicy{
+		applications:        []Application{},
 		positiveResultCache: cache.NewTimeCache(cachingPeriod),
 		negativeResultCache: cache.NewTimeCache(cachingPeriod),
 	}
 
-	_, remotePeerOperatorPublicKey, err := operator.GenerateKeyPair(local_v1.DefaultCurve)
+	_, peerOperatorPublicKey, err := operator.GenerateKeyPair(
+		local_v1.DefaultCurve,
+	)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	err = stakeMonitor.StakeTokens(remotePeerOperatorPublicKey)
+	err = policy.Validate(peerOperatorPublicKey)
+
+	expectedError := fmt.Errorf(
+		"remote peer has not been recognized by any application",
+	)
+	if !reflect.DeepEqual(expectedError, err) {
+		t.Errorf(
+			"unexpected error\nexpected: %v\nactual:   %v\n",
+			expectedError,
+			err,
+		)
+	}
```

Comment:
Done for the whole file in b72738ab3594b34b1ea4a1075ff07b23fba02511.

Reply:
Using `DeepEqual` to compare errors is not considered a good practice because we rely on the internal structure of errors. That's why we don't use `DeepEqual` in `testutils.AssertErrorsEqual`.

There are two options: compare strings or use [the new features](https://go.dev/blog/go1.13-errors) of errors. I personally prefer the latter here.

The first option would be:
```
expectedError := "could not validate if remote peer is recognized by application: [dummy error]"
testutils.AssertStringsEqual(t, "error", expectedError, err.Error())
```

The second option would be:
```
applicationError := fmt.Errorf("dummy error")

application1 := newMockApplication()
application1.setIsRecognized(peerOperatorPublicKey, result{
    isRecognized: false,
    err:          applicationError, // <-- here!
})

// (...)

if !errors.Is(err, applicationError) {
    t.Fatal("not the same")
}
```

We could even expose a testutil function for that:
```
func AssertAnyErrorInChainMatchesTarget(t *testing.T, target error, actual error) {
	if !errors.Is(actual, target) {
		t.Errorf(
			"no error in the chain matches the target\ntarget: %v\nactual:   %v\n",
			target,
			actual,
		)
	}
}
```

Note that this requires changing all `%v` to `%w` in `firewall.go` when creating errors.

Before:
```
return fmt.Errorf(
    "could not validate if remote peer is recognized by application: [%v]",
    err,
)
```

After:
```
return fmt.Errorf(
    "could not validate if remote peer is recognized by application: [%w]",
    err,
)
```

Reply:
Done in cb1f6f7ae6ea4c6cb39cd0f78d18f544de305a51:
- removed any changes I made to `testutils.AssertErrorsEqual`
- added `testutils.AssertAnyErrorInChainMatchesTarget`
- used `%w` instead of `%v` in `firewall.go`
- used `testutils.AssertAnyErrorInChainMatchesTarget` where needed. 
  Where it was not needed I used `testutils.AssertErrorsEqual`